Source/JavaScriptCore/inspector/remote/RemoteInspector.h

```diff
@@ -271,8 +271,8 @@ class JS_EXPORT_PRIVATE RemoteInspector final
     // from any thread.
     Lock m_mutex;
 
-    HashMap<TargetID, RemoteControllableTarget*> m_targetMap;
-    HashMap<TargetID, RefPtr<RemoteConnectionToTarget>> m_targetConnectionMap;
+    HashMap<TargetID, Ref<RemoteControllableTarget>> m_targetMap;
```

Comment:
I didn't check but please make sure you're not introducing a reference cycle here by switching from a raw pointer to a Ref. We usually convert raw pointers to WeakPtr/WeakRef to avoid cycles so you need to be sure.

Reply:
Great advice. It looks to me like it's safe to replace these raw pointers with `Ref`s—no child class of `RemoteControllableTarget` hold a strong `Ref` or `RefPtr` to this class (`RemoteInspector`) or `RemoteConnectionToTarget`.

- Though, there still exists a raw pointer in `WebAutomationSession` to a `FrontendChannel` [here](https://github.com/WebKit/WebKit/blob/main/Source/WebKit/UIProcess/Automation/WebAutomationSession.h#L371), which could potentially be a `RemoteConnectionToTarget`. If in the future we decide to convert that member variable to a smart pointer, we probably would have to make that a `WeakPtr` instead to avoid a ref cycle, but I believe making that variable weak is the correct thing to do (rather than making weak pointers here in `RemoteInspector`).

@patrickangle @burg Patrick and Blaze, can you help me verify that we're absolutely not creating a ref cycle with this change today?